src/Data/Text/Internal/Encoding/Utf8.hs

```diff
@@ -54,13 +54,15 @@ between :: Word8                -- ^ byte to check
 between x y z = x >= y && x <= z
 {-# INLINE between #-}
 
--- TODO make branchless by looking into Word64 by clz (ord c)
+-- This is a branchless version of
+-- utf8Length c
+--   | ord c < 0x80    = 1
+--   | ord c < 0x800   = 2
+--   | ord c < 0x10000 = 3
+--   | otherwise       = 4
 utf8Length :: Char -> Int
-utf8Length c
-  | ord c < 0x80    = 1
-  | ord c < 0x800   = 2
-  | ord c < 0x10000 = 3
-  | otherwise       = 4
+utf8Length (C# c) = I# ((1# +# geChar# c (chr# 0x80#)) +# (geChar# c (chr# 0x800#) +# geChar# c (chr# 0x10000#)))
```

Comment:
Do I get my name in a contributors file now? 😁 

More importantly, is there any benchmarking between this and the previous one? I think one is definitely clearer, but I would actually like to know if ti is faster or not.

Reply:
I briefly ran a couple of benchmarks, but did not notice statistically significant changes. Could be mostly because this function is not a bottleneck, but also note that (modulo casts, which should be no-ops) the previous implementation was also 6 machine instructions.